src/main/java/org/spongepowered/common/registry/SpongeCatalogRegistry.java

```diff
@@ -302,6 +308,22 @@ public SpongeCatalogRegistry registerRegistry(Class<CatalogType> catalogClass, C
         return this;
     }
 
+    private <T extends CatalogType> SpongeCatalogRegistry registerCallbackRegistry(Class<T> catalogClass, CatalogKey key, BiConsumer<ResourceLocation, T> callback) {
+        checkNotNull(catalogClass);
+        checkNotNull(key);
+
+        if (this.registries.get(key) != null) {
```

Comment:
Replace get + put with computeIfAbsent

Reply:
but it throws exception if present

Reply:
```
final CallbackRegistry<T> previous = this.registries.putIfAbsent(key, (Registry<CatalogType>) registry);
if (previous != null) {
    throw ...
}
```

(Same with compute, but putIfAbsent is fine since `registry` is used later and it doesn't need to be constructed lazily)